pkg/collector/corechecks/gpu/gpu.go

```diff
@@ -133,3 +172,20 @@ func (m *Check) Run() error {
 
 	return nil
 }
+
+func (m *Check) emitNvmlMetrics(snd sender.Sender) error {
+	for _, collector := range m.collectors {
+		log.Debugf("Collecting metrics from NVML collector: %s", collector.Name())
+		metrics, err := collector.Collect()
+		if err != nil {
+			log.Warnf("failed to collect metrics from NVML collector %s: %s", collector.Name(), err)
```

Comment:
better return the error and log it from the caller

Reply:
Fixed with a multierror so we don't stop all collectors if one fails.